examples/esp-chat-react/js/components/MessageSection.react.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+
+var MessageComposer = require('./MessageComposer.react');
+var MessageListItem = require('./MessageListItem.react');
+var React = require('react');
+
+var MessageSection = React.createClass({
+    componentDidMount: function () {
+        this._scrollToBottom();
+    },
+
+    componentDidUpdate: function() {
+        this._scrollToBottom();
+    },
+
+    _scrollToBottom: function() {
+        var ul = this.refs.messageList.getDOMNode();
+        ul.scrollTop = ul.scrollHeight;
+    },
+
+    render: function () {
+        var messageSection = this.props.model;
+        var messageListItems = messageSection.sortedMessages.map(function (message) {
+            return (
+                <MessageListItem
+                    key={message.id}
+                    model={message}
+                />
+            );
+        });
+        return (
+            <div className="message-section">
+                <h3 className="message-thread-heading">{messageSection.threadName}</h3>
+                <ul className="message-list" ref="messageList">
+                {messageListItems}
+                </ul>
+                <MessageComposer
+                    router={this.props.router}
```

Comment:
Yes. Reacts context API currently only exists in an undocumented form so it's not really stable enough to use in an example yet. There was some discussion on this in the other PR.